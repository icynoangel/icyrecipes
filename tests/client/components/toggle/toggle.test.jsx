import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import Toggle from './../../../../src/client/js/components/toggle/toggle';
import renderer from 'react-test-renderer';

describe('<Toggle />', function() {
  this.getComponent = (checked = false, disabled = false) => {
    return (
      <Toggle
        onClick={this.onClick}
        label={this.label}
        checked={checked}
        disabled={disabled}
      />
    );
  };

  beforeEach(() => {
    this.onClick = jest.fn();
    this.label = 'Toggle Button';
  });

  it('Should render <Toggle />', () => {
    const toggle = renderer.create(this.getComponent()).toJSON();
    expect(toggle).toMatchSnapshot();
  });

  it('Should render a checked <Toggle />', () => {
    const toggle = renderer.create(this.getComponent(!!'checked')).toJSON();
    expect(toggle).toMatchSnapshot();
  });

  it('Should render a disabled <Toggle />', () => {
    const toggle = renderer
      .create(this.getComponent(!'checked', !!'disabled'))
      .toJSON();
    expect(toggle).toMatchSnapshot();
  });

  it('Should render a checked and disabled <Toggle />', () => {
    const toggle = renderer
      .create(this.getComponent(!!'checked', !!'disabled'))
      .toJSON();
    expect(toggle).toMatchSnapshot();
  });

  it('Should change state when clicked and call onClick prop', () => {
    const wrapper = shallow(this.getComponent());
    wrapper.find('.toggle').simulate('click');
    expect(this.onClick).toHaveBeenCalledTimes(1);
    expect(wrapper.state('checked')).toBe(true);
  });

  it('Should not change state and not call onClick prop when clicked and is disabled', () => {
    const wrapper = shallow(this.getComponent(!'checked', !!'disabled'));
    wrapper.find('.toggle').simulate('click');
    expect(this.onClick).not.toHaveBeenCalled();
    expect(wrapper.state('checked')).toBe(false);
  });
});
