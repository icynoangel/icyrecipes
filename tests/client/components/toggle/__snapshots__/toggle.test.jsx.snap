// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> Should render <Toggle /> 1`] = `
<div
  className="toggle"
  onClick={[Function]}
>
  <div
    className="toggle__toggle-button"
  >
    <div
      className="toggle__toggle-button__slider"
    />
  </div>
  <div
    className="toggle__label"
  >
    Toggle
  </div>
</div>
`;

exports[`<Toggle /> Should render a checked <Toggle /> 1`] = `
<div
  className="toggle --checked"
  onClick={[Function]}
>
  <div
    className="toggle__toggle-button"
  >
    <div
      className="toggle__toggle-button__slider"
    />
  </div>
  <div
    className="toggle__label"
  >
    Toggle
  </div>
</div>
`;

exports[`<Toggle /> Should render a checked and disabled <Toggle /> 1`] = `
<div
  className="toggle --checked --disabled"
  onClick={[Function]}
>
  <div
    className="toggle__toggle-button"
  >
    <div
      className="toggle__toggle-button__slider"
    />
  </div>
  <div
    className="toggle__label"
  >
    Toggle
  </div>
</div>
`;

exports[`<Toggle /> Should render a disabled <Toggle /> 1`] = `
<div
  className="toggle --disabled"
  onClick={[Function]}
>
  <div
    className="toggle__toggle-button"
  >
    <div
      className="toggle__toggle-button__slider"
    />
  </div>
  <div
    className="toggle__label"
  >
    Toggle
  </div>
</div>
`;
